vms/platformvm/state/disk_staker_diff_iterator.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+const (
+	// startDiffKey = [subnetID] + [inverseHeight]
+	startDiffKeyLength = ids.IDLen + database.Uint64Size
+	// diffKey = [subnetID] + [inverseHeight] + [nodeID]
+	diffKeyLength       = startDiffKeyLength + ids.NodeIDLen
+	diffKeyHeightOffset = ids.IDLen
+	diffKeyNodeIDOffset = diffKeyHeightOffset + database.Uint64Size
+
+	// weightValue = [isNegative] + [weight]
+	weightValueLength = database.BoolSize + database.Uint64Size
+)
+
+var (
+	errUnexpectedDiffKeyLength     = fmt.Errorf("expected diff key length %d", diffKeyLength)
+	errUnexpectedWeightValueLength = fmt.Errorf("expected weight value length %d", weightValueLength)
+)
+
+// getStartDiffKey is used to determine the starting key when iterating.
+//
+// Note: the result should be a prefix of [getDiffKey] if called with the same
+// arguments.
+func getStartDiffKey(subnetID ids.ID, height uint64) []byte {
+	key := make([]byte, startDiffKeyLength)
+	copy(key, subnetID[:])
+	packIterableHeight(key[diffKeyHeightOffset:], height)
+	return key
+}
+
+func getDiffKey(subnetID ids.ID, height uint64, nodeID ids.NodeID) []byte {
+	key := make([]byte, diffKeyLength)
+	copy(key, subnetID[:])
+	packIterableHeight(key[diffKeyHeightOffset:], height)
+	copy(key[diffKeyNodeIDOffset:], nodeID[:])
+	return key
+}
+
+func parseDiffKey(key []byte) (ids.ID, uint64, ids.NodeID, error) {
+	if len(key) != diffKeyLength {
+		return ids.Empty, 0, ids.EmptyNodeID, errUnexpectedDiffKeyLength
+	}
+	var (
+		subnetID ids.ID
+		nodeID   ids.NodeID
+	)
+	copy(subnetID[:], key)
+	height := unpackIterableHeight(key[diffKeyHeightOffset:])
+	copy(nodeID[:], key[diffKeyNodeIDOffset:])
+	return subnetID, height, nodeID, nil
+}
+
+func getWeightValue(diff *ValidatorWeightDiff) []byte {
+	value := make([]byte, weightValueLength)
+	if diff.Decrease {
+		value[0] = database.BoolTrue
+	}
+	binary.BigEndian.PutUint64(value[database.BoolSize:], diff.Amount)
+	return value
+}
+
+func parseWeightValue(value []byte) (*ValidatorWeightDiff, error) {
```

Comment:
Similar to above, maybe `parseWeightDiff`?

Reply:
renamed.